### DISCUSSION TITLE
Node.js GOT Docker unix socket stream error

### DISCUSSION THREAD

**Discussion by bllmo** [sindresorhus/got#1887] [Q&A] [answered]
#### Describe the bug

 - Node.js version: v16.10.0
 - OS & version: macOS Big Sur 11.6

#### Actual behavior

Hello i have this code for attach a specific container with [GOT](https://www.npmjs.com/package/got) and retrive log stream (according to docker [documentation](https://docs.docker.com/engine/api/v1.41/#operation/ContainerAttach))

#### Code to reproduce

```js
const got = require("got");

(async () => {
  const stream = await got('unix:/var/run/docker.sock:/containers/efbb049e620f/attach', { 
    method: 'post',
    searchParam: {
      stream: true,
      stdin: true,
      stdout: true,
      stderr: true,
    },
    isStream: true
  });

  stream.on('data', console.log);
})();
```
but when i run this code i dont see any log in the console, how i can resolve this?

#### Checklist

- [ x] I have read the documentation.
- [ x] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment, marked as answer]
It doesn't show anything because you didn't send your data.

https://nodejs.org/api/stream.html#stream_writable_end_chunk_encoding_callback

**Comment by bllmo** [comment]
Hi @szmarczak, if i use [dockerode](https://github.com/apocas/dockerode) attach command and I run it with the same params I see docker container logs into the console.

![Screen Shot 2021-10-09 at 8 45 52 AM](https://user-images.githubusercontent.com/6557615/136647348-ad4419f9-a04b-4d14-9190-c6347619e5be.png)

I test my got code with httpbin socket endpoint and it works.

apparently, it doesn't work with unix sockets

**Comment by szmarczak** [comment]
I think you're doing something wrong. We have tests for unix sockets and they pass.

**Comment by bllmo** [comment]
resolved! I put an empty json object to the request.

```
const got = require("got");

(async () => {
  const stream = await got('unix:/var/run/docker.sock:/containers/efbb049e620f/attach', { 
    method: 'post',
    json: {},
    searchParam: {
      stream: true,
      stdin: true,
      stdout: true,
      stderr: true,
    },
    isStream: true
  });

  stream.on('data', console.log);
})();
```
